client/state-db/src/lib.rs

```diff
@@ -237,8 +237,7 @@ impl<BlockHash: Hash + MallocSizeOf, Key: Hash + MallocSizeOf> StateDbSync<Block
 		);
 		match &db_mode {
 			Some(v) if v.as_slice() == mode.id() => Ok(()),
-			Some(v) => Err(Error::InvalidPruningMode(String::from_utf8_lossy(v).into())),
-			None => Ok(()),
+			_ => Ok(()),
```

Comment:
I've no idea what I've done here 😄 I just made it work. But I am fairly sure that it is not a big deal and there will be a good solution for it. My issue was that I was getting the `InvalidPruningMode` error here while the new subcommand should have nothing to do with this actually. I guess there should be a way to tell the state-db to remain calm here.

Reply:
This means you have mixed different pruning modes. Did you run it as validator and later without the --validator flag? Whatever you do, please revert this ;)

Reply:
yes this is a no-go, I just wanted to unblock myself.

Reply:
Fixed, now I won't create partial components any more, only a manual task manager.